src/components/ha-selector/ha-selector-color-rgb.ts

```diff
@@ -0,0 +1,93 @@
+import { mdiPalette } from "@mdi/js";
+import { css, html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { HomeAssistant } from "../../types";
+import { IconSelector } from "../../data/selector";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../ha-color-picker";
+import "../ha-icon-button";
+
+@customElement("ha-selector-color-rgb")
+export class HaColorRGBSelector extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property() public selector!: IconSelector;
+
+  @property() public value?: string;
+
+  @property() public label?: string;
+
+  @property({ type: Boolean, reflect: true }) public disabled = false;
+
+  @state() private _saturationSegments = 8;
+
+  @state() private _hueSegments = 24;
+
+  protected render() {
+    return html`
+      <div class="segmentationContainer">
+        <ha-color-picker
+          class="color"
+          throttle="500"
+          .label=${this.label}
+          .desiredRgbColor=${this.value}
+          .hueSegments=${this._hueSegments}
+          .saturationSegments=${this._saturationSegments}
+          @colorselected=${this._valueChanged}
+        >
+        </ha-color-picker>
+        <ha-icon-button
+          .path=${mdiPalette}
+          @click=${this._segmentClick}
+          class="segmentationButton"
+        ></ha-icon-button>
+      </div>
+    `;
+  }
+
+  private _valueChanged(ev: CustomEvent) {
+    fireEvent(this, "value-changed", {
+      value: [ev.detail.rgb.r, ev.detail.rgb.g, ev.detail.rgb.b],
```

Comment:
Did we decide what the best return result for this would be for the backend?

Reply:
I would make it an object, so maybe it can be extended in the future.

Reply:
List is fine.